src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -31,6 +34,29 @@
 from promptflow.storage import AbstractRunStorage
 
 
+def signal_handler(signum, frame):
+    signame = signal.Signals(signum).name
+    logger.info("Execution stopping. Handling signal %s (%s)", signame, signum)
+    try:
+        process = psutil.Process(os.getpid())
+        children = process.children(recursive=True)
+        for child in children:
+            try:
+                child.terminate()
+                logger.info("Successfully terminated child process with pid %s", child.pid)
+            except NoSuchProcess:
+                logger.warning("Process %s already terminated and not found, skipping", process.pid)
+        logger.info("Successfully terminated process with pid %s", process.pid)
+        process.terminate()
+    except Exception:
+        logger.warning("Error when handling execution stop signal", exc_info=True)
+    finally:
+        sys.exit(1)
+
+
+signal.signal(signal.SIGINT, signal_handler)
```

Comment:
Can we do this inside some function? It is not a good practice to do stuffs in the module level.

Reply:
Add `signal.signal(signal.SIGINT, signal_handler)` to function level. 
Thank you for your advice.